src/cli-parser.ts

```diff
@@ -0,0 +1,57 @@
+import path from "node:path";
+
+import { program } from "commander";
+
+import { Ef } from "./common";
+import { logErrorAndExit } from "./logger";
+
+type CheckOnlyOptionType = null | {
+  section: string;
+  contract?: string;
+  checksType?: string;
+  method?: string;
+  schemas?: string;
+};
+
+export function parseCmdLineArguments() {
+  program
+    .argument("<config-path>", "path to .yaml state config file")
+    .allowExcessArguments(false)
+    .option(
+      "-o, --only <check-path>",
+      `only checks to do, e.g. 'l2/proxyAdmin/${Ef.checks}/owner', 'l1', 'l1/controller'`,
+    )
+    .option("--generate", "generate a populated config from the seed one")
+    .option("--abi", "check that the saved ABIs and explorer ABIs are equal")
+    .option("--schemas", "generate main and seed JSON-schemas")
```

Comment:
```suggestion
    .option("--update-abi", "download all ABIs replacing existing files")
```

`schemas` should live under a separate script being orthogonal to the main flow of this file (as commented above)